### PR TITLE
Use CSS color vars with fallback in legacy Primitive tokens

### PR DESCRIPTION
**Pull request by langermank**
Closes https://github.com/github/primer/issues/2987

As a followup to moving the legacy theme files into Primer React from Primer Primitives, this PR introduces the new v8 color CSS vars with fallbacks as the _value_ of the JS variables. This allows us to staff ship v8 variables without rewriting all of our CSS upfront. It also may help with the [theme flashing issue](https://github.com/github/primer/issues/2927).

### Code review

This looks like a lot of changes (it is) but the thing we are most concerned about is if the **raw value** has changes. The easiest way to see that is to primarily review the lefthand side to see which values have changed. There are a few intentional changes:

- sometimes rgba transparent becomes `transparent`
- the fg muted color was updated in a newer Primitives release which is reflected here
- random bug fixes

diff view

### Changelog

- Adds CSS variable color values to the legacy Primitive design tokens.
- Three values include: the new v8 color CSS variable, a fallback of the old color CSS variable, and the raw value.

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Testing & Reviewing

We'll definitely want to test this against dotcom with an integration test PR.

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: src/legacy-theme/ts/colors/dark.ts [hunk 1]
```diff
@@ -1,503 +1,532 @@
...
+      lineBg: 'var(--diffBlob-deletion-bgColor-line, var(--color-diff-blob-deletion-line-bg, rgba(248,81,73,0.1)))',
+      wordBg: 'var(--diffBlob-deletion-bgColor-word, var(--color-diff-blob-deletion-word-bg, rgba(248,81,73,0.4)))',
     },
     hunk: {
-      numBg: 'rgba(56,139,253,0.4)',
+      numBg: 'var(--diffBlob-hunk-bgColor-num, var(--color-diff-blob-hunk-num-bg, rgba(56,139,253,0.4)))',
     },
     expander: {
-      icon: '#7d8590',
+      icon: 'var(--diffBlob-expander-iconColor, var(--color-diff-blob-expander-icon, #848d97))',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:27]:
```suggestion
      icon: 'var(--diffBlob-expander-iconColor, var(--color-diff-blob-expander-icon, #7d8590))',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 2]
```diff
@@ -1,503 +1,532 @@
...
-    loglineWarningText: '#7d8590',
-    loglineWarningNumText: '#d29922',
-    loglineWarningBg: 'rgba(187,128,9,0.15)',
-    loglineCommandText: '#2f81f7',
-    loglineSectionText: '#3fb950',
+    bg: 'var(--color-checks-bg, #010409)',
+    runBorderWidth: 'var(--color-checks-run-border-width, 1px)',
+    containerBorderWidth: 'var(--color-checks-container-border-width, 1px)',
+    textPrimary: 'var(--color-checks-text-primary, #e6edf3)',
+    textSecondary: 'var(--color-checks-text-secondary, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:104]:
```suggestion
    textSecondary: 'var(--color-checks-text-secondary, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 3]
```diff
@@ -1,503 +1,532 @@
...
+    textLink: 'var(--color-checks-text-link, #2f81f7)',
+    btnIcon: 'var(--color-checks-btn-icon, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:106]:
```suggestion
    btnIcon: 'var(--color-checks-btn-icon, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 4]
```diff
@@ -1,503 +1,532 @@
...
+    btnHoverIcon: 'var(--color-checks-btn-hover-icon, #e6edf3)',
+    btnHoverBg: 'var(--color-checks-btn-hover-bg, rgba(110,118,129,0.1))',
+    inputText: 'var(--color-checks-input-text, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:109]:
```suggestion
    inputText: 'var(--color-checks-input-text, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 5]
```diff
@@ -1,503 +1,532 @@
...
+    dropdownText: 'var(--color-checks-dropdown-text, #e6edf3)',
+    dropdownBg: 'var(--color-checks-dropdown-bg, #161b22)',
+    dropdownBorder: 'var(--color-checks-dropdown-border, #30363d)',
+    dropdownShadow: 'var(--color-checks-dropdown-shadow, rgba(1,4,9,0.3))',
+    dropdownHoverText: 'var(--color-checks-dropdown-hover-text, #e6edf3)',
+    dropdownHoverBg: 'var(--color-checks-dropdown-hover-bg, rgba(110,118,129,0.1))',
+    dropdownBtnHoverText: 'var(--color-checks-dropdown-btn-hover-text, #e6edf3)',
+    dropdownBtnHoverBg: 'var(--color-checks-dropdown-btn-hover-bg, rgba(110,118,129,0.1))',
+    scrollbarThumbBg: 'var(--color-checks-scrollbar-thumb-bg, rgba(110,118,129,0.4))',
+    headerLabelText: 'var(--color-checks-header-label-text, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:127]:
```suggestion
    headerLabelText: 'var(--color-checks-header-label-text, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 6]
```diff
@@ -1,503 +1,532 @@
...
+    headerLabelOpenText: 'var(--color-checks-header-label-open-text, #e6edf3)',
+    headerBorder: 'var(--color-checks-header-border, #21262d)',
+    headerIcon: 'var(--color-checks-header-icon, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:130]:
```suggestion
    headerIcon: 'var(--color-checks-header-icon, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 7]
```diff
@@ -1,503 +1,532 @@
...
+    lineText: 'var(--color-checks-line-text, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:131]:
```suggestion
    lineText: 'var(--color-checks-line-text, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 8]
```diff
@@ -1,503 +1,532 @@
...
+    lineNumText: 'var(--color-checks-line-num-text, #6e7681)',
+    lineTimestampText: 'var(--color-checks-line-timestamp-text, #6e7681)',
+    lineHoverBg: 'var(--color-checks-line-hover-bg, rgba(110,118,129,0.1))',
+    lineSelectedBg: 'var(--color-checks-line-selected-bg, rgba(56,139,253,0.1))',
+    lineSelectedNumText: 'var(--color-checks-line-selected-num-text, #2f81f7)',
+    lineDtFmText: 'var(--color-checks-line-dt-fm-text, #ffffff)',
+    lineDtFmBg: 'var(--color-checks-line-dt-fm-bg, #9e6a03)',
+    gateBg: 'var(--color-checks-gate-bg, rgba(187,128,9,0.15))',
+    gateText: 'var(--color-checks-gate-text, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:140]:
```suggestion
    gateText: 'var(--color-checks-gate-text, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 9]
```diff
@@ -1,503 +1,532 @@
...
+    gateWaitingText: 'var(--color-checks-gate-waiting-text, #d29922)',
+    stepHeaderOpenBg: 'var(--color-checks-step-header-open-bg, #161b22)',
+    stepErrorText: 'var(--color-checks-step-error-text, #f85149)',
+    stepWarningText: 'var(--color-checks-step-warning-text, #d29922)',
+    loglineText: 'var(--color-checks-logline-text, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:145]:
```suggestion
    loglineText: 'var(--color-checks-logline-text, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 10]
```diff
@@ -1,503 +1,532 @@
...
+    loglineNumText: 'var(--color-checks-logline-num-text, #6e7681)',
+    loglineDebugText: 'var(--color-checks-logline-debug-text, #a371f7)',
+    loglineErrorText: 'var(--color-checks-logline-error-text, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:148]:
```suggestion
    loglineErrorText: 'var(--color-checks-logline-error-text, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 11]
```diff
@@ -1,503 +1,532 @@
...
+    loglineErrorNumText: 'var(--color-checks-logline-error-num-text, #6e7681)',
+    loglineErrorBg: 'var(--color-checks-logline-error-bg, rgba(248,81,73,0.1))',
+    loglineWarningText: 'var(--color-checks-logline-warning-text, #848d97)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:151]:
```suggestion
    loglineWarningText: 'var(--color-checks-logline-warning-text, #7d8590)',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 12]
```diff
@@ -1,503 +1,532 @@
...
-    activeBg: 'rgba(110,118,129,0.4)',
-    disabledBg: '#21262d',
-    fg: '#7d8590',
-    disabledFg: '#010409',
-    border: 'rgba(0,0,0,0)',
+    bg: 'var(--controlTrack-bgColor-rest, var(--color-switch-track-bg, rgba(110,118,129,0.1)))',
+    hoverBg: 'var(--controlTrack-bgColor-hover, var(--color-switch-track-hover-bg, hsla(215,8%,72%,0.1)))',
+    activeBg: 'var(--controlTrack-bgColor-active, var(--color-switch-track-active-bg, rgba(110,118,129,0.4)))',
+    disabledBg: 'var(--controlTrack-bgColor-disabled, var(--color-switch-track-disabled-bg, #21262d))',
+    fg: 'var(--controlTrack-fgColor-rest, var(--color-switch-track-fg, #848d97))',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:385]:
```suggestion
    fg: 'var(--controlTrack-fgColor-rest, var(--color-switch-track-fg, #7d8590))',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 13]
```diff
@@ -1,503 +1,532 @@
...
   },
   treeViewItem: {
     chevron: {
-      hoverBg: 'rgba(177,186,196,0.12)',
+      hoverBg:
+        'var(--control-transparent-bgColor-hover, var(--color-tree-view-item-chevron-hover-bg, rgba(177,186,196,0.12)))',
     },
     directory: {
-      fill: '#7d8590',
+      fill: 'var(--treeViewItem-leadingVisual-bgColor-rest, var(--color-tree-view-item-directory-fill, #848d97))',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:429]:
```suggestion
      fill: 'var(--treeViewItem-leadingVisual-bgColor-rest, var(--color-tree-view-item-directory-fill, #7d8590))',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 14]
```diff
@@ -1,503 +1,532 @@
...
     },
   },
   fg: {
-    default: '#e6edf3',
-    muted: '#7d8590',
-    subtle: '#6e7681',
-    onEmphasis: '#ffffff',
+    default: 'var(--fgColor-default, var(--color-fg-default, #e6edf3))',
+    muted: 'var(--fgColor-muted, var(--color-fg-muted, #848d97))',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:434]:
```suggestion
    muted: 'var(--fgColor-muted, var(--color-fg-muted, #7d8590))',
```

File: src/legacy-theme/ts/colors/dark.ts [hunk 15]
```diff
@@ -1,503 +1,532 @@
...
+      lineBg: 'var(--diffBlob-deletion-bgColor-line, var(--color-diff-blob-deletion-line-bg, rgba(248,81,73,0.1)))',
+      wordBg: 'var(--diffBlob-deletion-bgColor-word, var(--color-diff-blob-deletion-word-bg, rgba(248,81,73,0.4)))',
     },
     hunk: {
-      numBg: 'rgba(56,139,253,0.4)',
+      numBg: 'var(--diffBlob-hunk-bgColor-num, var(--color-diff-blob-hunk-num-bg, rgba(56,139,253,0.4)))',
     },
     expander: {
-      icon: '#7d8590',
+      icon: 'var(--diffBlob-expander-iconColor, var(--color-diff-blob-expander-icon, #7d8590))',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:27]:
```suggestion
      icon: 'var(--diffBlob-expander-iconColor, var(--color-diff-blob-expander-icon, #848d97))',
```

File: src/legacy-theme/ts/colors/dark.ts (hunk 1 above)
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:27]: Note that these changes from `7d8590` to `848d97` are actually intended and correct ✅

File: src/legacy-theme/ts/colors/dark.ts [hunk 16]
```diff
@@ -1,503 +1,532 @@
...
-    textSecondary: '#7d8590',
-    textLink: '#2f81f7',
-    btnIcon: '#7d8590',
-    btnHoverIcon: '#e6edf3',
-    btnHoverBg: 'rgba(110,118,129,0.1)',
-    inputText: '#7d8590',
-    inputPlaceholderText: '#6e7681',
-    inputFocusText: '#e6edf3',
-    inputBg: '#161b22',
-    inputShadow: '0 0 0 1px (obj) => (0, get_1.default)(obj, path)',
```
Comment (langermank) [inline src/legacy-theme/ts/colors/dark.ts:110]: This has been broken for awhile, I replaced it with the correct value

**Review by joshblack — APPROVED** [review]
LGTM! 🥳 I had two broader questions as it relates to this PR:

- Is there any potential impact as it relates to how folks use `themeGet`? It seems like most people just use it to get a value versus interacting with it so seems like it should be good?
- What's the best way for us to test this out in dotcom to make sure we're confident no styling issues make their way into a release?

**Pull request merged**